[BUG] test_hash_agg_with_nan_keys failed with a DATAGEN_SEED=1702335559

**Describe the bug**

```
FAILED ../../src/main/python/hash_aggregate_test.py::test_hash_agg_with_nan_keys[[('a', RepeatSeq(Double)), ('b', Float), ('c', Long)]][DATAGEN_SEED=1702335559, INJECT_OOM, IGNORE_ORDER, APPROXIMATE_FLOAT] - AssertionError: GPU and CPU float values are different [17, 'sum_of_bees']
```

It was running against Spark 3.2.4.

I tried to reproduce this locally and I could not. The only explanation I have is that the GPU did the sum in a non-deterministic order and we some how ended up with a number that was larger than approximate float could handle.  The key (a) on line 17 was  
`-3.1261173254359074e-295` and the values were `1.7718319043726909e+25` vs `1.7756097975589866e+25`.

Not sure what more to do except dig in a little deeper to see what might have caused the result by looking at the values used by the SUM.

When I ran this in a loop 100 times, I got it to fail just once:
```
[gw1] [100%] FAILED ../../src/main/python/hash_aggregate_test.py::test_hash_agg_with_nan_keys[[('a', RepeatSeq(Double)), ('b', Float), ('c', Long)]][DATAGEN_SEED=1702335559, INJECT_OOM, IGNORE_ORDER, APPROXIMATE_FLOAT]

cpu = 1.7718319043726909e+25, gpu = 1.7726637916678512e+25
path = [17, 'sum_of_bees']

-Row(a=-3.1261173254359074e-295, count_stars=21, count_parameterless=0, count_bees=20, sum_of_bees=1.7718319043726909e+25, max_seas=7098372324280358315, min_seas=-9170459960093196357, count_distinct_cees=20, average_seas=4.216639238479068e+17)
+Row(a=-3.1261173254359074e-295, count_stars=21, count_parameterless=0, count_bees=20, sum_of_bees=1.7726637916678512e+25, max_seas=7098372324280358315, min_seas=-9170459960093196357, count_distinct_cees=20, average_seas=4.216639238479067e+17)
```
Need to capture the contents of the b columns.

--- a/integration_tests/src/main/python/hash_aggregate_test.py
+++ b/integration_tests/src/main/python/hash_aggregate_test.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2020-2023, NVIDIA CORPORATION.
+# Copyright (c) 2020-2024, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.

--- a/integration_tests/src/main/python/hash_aggregate_test.py
+++ b/integration_tests/src/main/python/hash_aggregate_test.py
@@ -159,7 +159,7 @@ _nan_zero_double_special_cases = [
 
 _grpkey_doubles_with_nan_zero_grouping_keys = [
     ('a', RepeatSeqGen(DoubleGen(nullable=(True, 10.0), special_cases=_nan_zero_double_special_cases), length=50)),
-    ('b', FloatGen(nullable=(True, 10.0))),
+    ('b', IntegerGen(nullable=(True, 10.0))),
     ('c', LongGen())]
 
 # Schema for xfail cases
@@ -1154,7 +1154,6 @@ def test_hash_multiple_filters(data_gen, conf):
         'min(a), max(b) filter (where c > 250) from hash_agg_table group by a',
         conf)
 
-@datagen_overrides(seed=0, reason='https://github.com/NVIDIA/spark-rapids/issues/10026')
 @approximate_float
 @ignore_order
 @pytest.mark.parametrize('data_gen', [_grpkey_floats_with_nan_zero_grouping_keys,
@@ -1222,7 +1221,7 @@ def test_hash_agg_with_struct_of_array_fallback(data_gen):
 
 @approximate_float
 @ignore_order
-@pytest.mark.parametrize('data_gen', [ _grpkey_doubles_with_nan_zero_grouping_keys], ids=idfn)
+@pytest.mark.parametrize('data_gen', [ _grpkey_floats_with_nulls_and_nans ], ids=idfn)
 def test_count_distinct_with_nan_floats(data_gen):
     assert_gpu_and_cpu_are_equal_sql(
         lambda spark : gen_df(spark, data_gen, length=1024),

